dotCMS/src/main/java/com/dotmarketing/image/filter/CropImageFilter.java

```diff
@@ -40,21 +49,60 @@ public File runFilter(File file,  Map<String, String[]> parameters) {
 		BufferedImage src;
 		try {
 			src = ImageIO.read(file);
-			if(x > src.getWidth() || y > src.getHeight()){
-				return file;
-				
-			}
 			
-			if(x + w > src.getWidth()){
+			Dimension current = new Dimension(src.getWidth(), src.getHeight());
+
+			
+	        if(wIn ==0 && hIn >0){
+	            h = Math.round(hIn <=1  ? current.height * hIn : hIn);
+	            w = Math.round(h * current.width / current.height);
+	        }
+	        else if(wIn >0 && hIn ==0){
+	            w = Math.round(wIn <= 1 ? current.width * wIn : wIn);
+	            h = Math.round(w * current.height / current.width);
+	        }
+            else if(wIn >0 && hIn >0){
+                w = Math.round(wIn <= 1 ? current.width * wIn : wIn);
+                h = Math.round(hIn <= 1 ? current.height * hIn : hIn);
+            }
+	        else{
+	            w = current.width;
+	            h = current.height;
+	        }
+	        
+			
+	        if(x > current.getWidth() || y > current.getHeight()){
+	            return file;   
+	        }
+	        
+		    Optional<Point> centerOpt = (x==0 && y==0) ? calcFocalPoint(src, parameters)  : Optional.empty();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'centerOpt' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4761083)